localstack/utils/aws/aws_stack.py

```diff
@@ -413,7 +413,6 @@ def fix_account_id_in_arns(response, colon_delimiter=":", existing=None, replace
 
 
 def inject_test_credentials_into_env(env):
-    env = env or {}
```

Comment:
Minor nitpick: we could hedge against `None` parameters here (this was likely the original intention of this faulty line):
```
if not env:
    return
```
(shouldn't hold back the merge, though - we'll add that in a future change 👍 )

Reply:
True, but given the calling code, I don't see `env` being ever None.

_BTW, a better implementation for all the related code wouldn't mutate the argument and return void/unit. Rather, they'd take the existing env as a param, merge it with the extra values it wants to inject, and return a new environment. Then those functions could be chained to a resulting value so even the outer `env` doesn't need to be mutated._

Reply:
> BTW, a better implementation for all the related code wouldn't mutate the argument and return void/unit. Rather, they'd take the existing env as a param, merge it with the extra values it wants to inject, and return a new environment. Then those functions could be chained to a resulting value so even the outer env doesn't need to be mutated.

Good point. Agreed - a functional pattern without side effects would be preferable here (and in general). 👍